api/bindings.h

```diff
@@ -144,14 +144,86 @@ Buffer handle(cache_t *cache,
+                         Buffer env,
+                         Buffer msg,
+                         DB db,
+                         GoApi api,
+                         GoQuerier querier,
+                         uint64_t gas_limit,
+                         bool print_debug,
+                         uint64_t *gas_used,
+                         Buffer *err);
+
+Buffer ibc_channel_connect(cache_t *cache,
+                           Buffer code_id,
+                           Buffer env,
+                           Buffer msg,
+                           DB db,
+                           GoApi api,
+                           GoQuerier querier,
+                           uint64_t gas_limit,
+                           bool print_debug,
+                           uint64_t *gas_used,
+                           Buffer *err);
+
+Buffer ibc_channel_open(cache_t *cache,
+                        Buffer code_id,
+                        Buffer env,
+                        Buffer msg,
+                        DB db,
+                        GoApi api,
+                        GoQuerier querier,
+                        uint64_t gas_limit,
+                        bool print_debug,
+                        uint64_t *gas_used,
+                        Buffer *err);
+
+Buffer ibc_packet_ack(cache_t *cache,
+                      Buffer code_id,
+                      Buffer env,
+                      Buffer msg,
+                      DB db,
+                      GoApi api,
+                      GoQuerier querier,
+                      uint64_t gas_limit,
+                      bool print_debug,
+                      uint64_t *gas_used,
+                      Buffer *err);
+
+Buffer ibc_packet_receive(cache_t *cache,
+                          Buffer code_id,
+                          Buffer env,
+                          Buffer msg,
+                          DB db,
+                          GoApi api,
+                          GoQuerier querier,
+                          uint64_t gas_limit,
+                          bool print_debug,
+                          uint64_t *gas_used,
+                          Buffer *err);
+
+Buffer ibc_packet_timeout(cache_t *cache,
+                          Buffer code_id,
+                          Buffer env,
+                          Buffer msg,
+                          DB db,
+                          GoApi api,
+                          GoQuerier querier,
+                          uint64_t gas_limit,
+                          bool print_debug,
+                          uint64_t *gas_used,
+                          Buffer *err);
+
 cache_t *init_cache(Buffer data_dir,
                     Buffer supported_features,
                     uint32_t cache_size,
                     uint32_t instance_memory_limit,
                     Buffer *err);
 
 Buffer instantiate(cache_t *cache,
-                   Buffer contract_id,
+                   Buffer code_id,
```

Comment:
Can't we change this to uint64, which is a significantly simpler data type (independent of this PR)?

Reply:
Good idea, let's check with Alex on the wasmd side for what changes that would bring.

Reply:
Turns out this is the cheksum (https://github.com/CosmWasm/wasmvm/issues/170)

Reply:
ja the hash. We store it in code info and pass it for instantiation here. We can also store some other identifier returned from the `store` function but hash is quite useful for users to verify their code upload is consistent.

Reply:
The checksum is also the better identifier for communication with the cache, which does not care about the meta data and storage model of the chain. It is really just a naming issue for historic reasons.

Reply:
Addressed in #171